finagle-core/src/main/scala/com/twitter/finagle/service/RateLimitingFilter.scala

```diff
@@ -13,20 +13,23 @@ class LocalRateLimitingStrategy[Req](categorizer: Req => String, windowSize: Dur
 
   private[this] val rates = mutable.HashMap.empty[String, (Int, Time)]
 
-  def apply(req: Req) = synchronized {
+  def apply(req: Req): Future[Boolean] = synchronized {
     val now = Time.now
     val id = categorizer(req)
     val (remainingRequests, timestamp) = rates.getOrElse(id, (rate, now))
 
-    val accept = if (timestamp.until(now) > windowSize) {
-      rates(id) = (rate, now)
+    val accept = if (rate <= 0) {
```

Comment:
Negative (or zero) `rate` is a misconfiguration in my books. Maybe we can just fail fast early, at the `LocalRateLimitingStrategy ` construction time?

Reply:
Ok, I added patch so that `LocalRateLimitingStrategy` fails at construction time.

Reply:
👍